scripts/get-all-images.py

```diff
@@ -0,0 +1,245 @@
+logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+log = logging.getLogger(__name__)
+
+# consts
+EXCLUDE_CHARMS = ["kubeflow-roles"]
+
+GH_REPO_KEY = "_github_repo_name"
+GH_BRANCH_KEY = "_github_repo_branch"
+GH_DEPENDENCY_REPO_KEY = "_github_dependency_repo_name"
+GH_DEPENDENCY_BRANCH_KEY = "_github_dependency_repo_branch"
+GET_IMAGES_SH = "tools/get-images.sh"
+
+
+def is_dependency_app(app: dict) -> bool:
+    """
+    Return True if app in bundle is not owned by Analytics team, False
+    otherwise.
+
+    Args:
+        app(dict): app metadata from a bundle.yaml in dictionary form
+
+    Returns:
+        True if app has GH dependency metadata, else False
+    """
+    if GH_DEPENDENCY_REPO_KEY in app and GH_DEPENDENCY_BRANCH_KEY in app:
+        return True
+
+    return False
+
+
+def bundle_app_contains_gh_metadata(app: dict) -> bool:
+    """
+    Given an application in a bundle check if it contains github metadata keys
+    to be able to be parsed properly.
+
+    Args:
+        app(dict): app metadata from a bundle.yaml in dictionary form
+
+    Returns:
+        True if app has GH metadata, else False
+    """
+    if is_dependency_app(app):
+        return True
+
+    if GH_REPO_KEY in app and GH_BRANCH_KEY in app:
+        return True
+
+    return False
+
+
+def clone_repo(repo_name: str, branch: str):
+    """
+    Clones locally a repo and returns the path of the folder created.
+
+    Args:
+        repo_name(str): name of the repo to clone
+        branch(str): branch to checkout to, once cloned the repo
+    """
+    repo_url = f"https://github.com/canonical/{repo_name}.git"
+
+    logging.info(f"Cloning repo {repo_url}")
+    repo = git.Repo.clone_from(repo_url, str(repo_name))
+
+    logging.info(f"Checking out to branch {branch}")
+    repo.git.checkout(branch)
+
+
+def get_analytics_app_images(app: dict) -> list[str]:
+    """
+    This function gets the images used by a charm developed by us, by:
+    1. Cloning the repo of the charm
+    2. Running the repo's tools/get-images.sh
+    3. Delete the repo
+
+    If the tools/get-images.sh of a repo fails for any reason then this
+    script will also fail.
+    """
+    images = []
+    repo_dir = app[GH_REPO_KEY]
+    clone_repo(app[GH_REPO_KEY], app[GH_BRANCH_KEY])
+
+    logging.info(f"Executing repo's {GET_IMAGES_SH} script")
+    try:
+        process = subprocess.run(["bash", "tools/get-images.sh"],
+                                 cwd=repo_dir, capture_output=True,
+                                 text=True, check=True)
+    except subprocess.CalledProcessError as exc:
+        logging.error("Script '%s' for charm '%s' failed: %s",
+                      GET_IMAGES_SH, app["charm"], exc.stderr)
+        raise exc
+    finally:
+        # cleanup
+        shutil.rmtree(repo_dir)
+
+    images = process.stdout.split("\n")
+
+    logging.info("Found the following images:")
+    [logging.info(image) for image in images]
```

Comment:
```suggestion
    [logging.info('* '+image) for image in images]
```
how about something like that for readability?

Reply:
Agree, that's a good idea.

While working on this though I realised that the script was including an empty string as image, which I saw because after adding the `* ` then there was an entry with only the asterisk.

I fixed it by `strip()`-ing the input. I also converted to a regular for loop since I understand it seems a bit exotic :)

ae4c71e